services/alertmanager/alertmanager.go

```diff
@@ -157,6 +169,159 @@ func (svc *Service) updateConfiguration(ctx context.Context) {
 	svc.l.Infof("%s created", alertmanagerConfigPath)
 }
 
+func (svc *Service) populateConfig(cfg *alertmanager.Config) error {
+	var settings *models.Settings
+	var rules []*models.Rule
+	var channels []*models.Channel
+	e := svc.db.InTransaction(func(tx *reform.TX) error {
+		var err error
+		settings, err = models.GetSettings(tx.Querier)
+		if err != nil {
+			return err
+		}
+
+		rules, err = models.FindRules(tx.Querier)
+		if err != nil {
+			return err
+		}
+
+		channels, err = models.FindChannels(tx.Querier)
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+	if e != nil {
+		return errors.Errorf("Failed to fetch items from database: %s", e)
+	}
+
+	cfg.Global = &alertmanager.GlobalConfig{}
+	if settings.IntegratedAlerting.EmailAlertingSettings != nil {
+		cfg.Global.SMTPFrom = settings.IntegratedAlerting.EmailAlertingSettings.From
+		cfg.Global.SMTPHello = settings.IntegratedAlerting.EmailAlertingSettings.Hello
+		cfg.Global.SMTPAuthIdentity = settings.IntegratedAlerting.EmailAlertingSettings.Identity
+		cfg.Global.SMTPAuthUsername = settings.IntegratedAlerting.EmailAlertingSettings.Username
+		cfg.Global.SMTPAuthPassword = settings.IntegratedAlerting.EmailAlertingSettings.Password
+		cfg.Global.SMTPAuthSecret = settings.IntegratedAlerting.EmailAlertingSettings.Secret
```

Comment:
How to handle case when user already specified these values in base config file?

Reply:
For now: log a warning about it and overwrite.

In the the future, we might want to move those settings into each receiver

Reply:
i forgot that the base file could be edited, in that case I think we should preserve the changes that the user made in the base file, @AlekSi can you please confirm this?

Reply:
Nope, see my comment above

Reply:
added warning